bundle/CMakeLists.txt

```diff
@@ -0,0 +1,147 @@
+# ------------------------------------------------------------------------- #
+# JEDI GDAS Bundle             #
+# ------------------------------------------------------------------------- #
+
+# Check for minimim cmake requirement
+cmake_minimum_required( VERSION 3.20 FATAL_ERROR )
+
+find_package(ecbuild 3.5 REQUIRED HINTS ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/../ecbuild)
+
+project(GDAS-bundle VERSION 1.0.0 LANGUAGES C CXX Fortran )
+
+include(GNUInstallDirs)
+enable_testing()
+
+# Build type.
+if(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo|MinSizeRel)$")
+  message(STATUS "Setting build type to 'Release' as none was specified.")
+  set(CMAKE_BUILD_TYPE
+      "Release"
+      CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release"
+                                               "MinSizeRel" "RelWithDebInfo")
+endif()
+
+# Find dependencies.
+find_package(Python3 REQUIRED COMPONENTS Interpreter)
+
+# Include ecbuild_bundle macro
+include( ecbuild_bundle )
+
+# Enable MPI
+set( ENABLE_MPI ON CACHE BOOL "Compile with MPI" )
+
+# Handle user options.
+option(BUILD_GDASBUNDLE "Build GDAS Bundle" ON)
+option(CLONE_JCSDADATA "Clone JCSDA test data repositories" OFF)
+option(WORKFLOW_TESTS "Include global-workflow dependent tests" OFF)
+
+# Initialize bundle
+# -----------------
+ecbuild_bundle_initialize()
+
+# Build bundle source code.
+if(BUILD_GDASBUNDLE)
+
+# jedi-cmake
+  ecbuild_bundle( PROJECT jedicmake SOURCE "../sorc/jedicmake" )
+  include( jedicmake/cmake/Functions/git_functions.cmake )
+
+# ECMWF libraries
+  option("BUNDLE_SKIP_ECKIT" "Don't build eckit" "ON" ) # Skip eckit build unless user passes -DBUNDLE_SKIP_ECKIT=OFF
+  option("BUNDLE_SKIP_FCKIT" "Don't build fckit" "ON") # Skip fckit build unless user passes -DBUNDLE_SKIP_FCKIT=OFF
+  option("BUNDLE_SKIP_ATLAS" "Don't build atlas" "ON") # Skip atlas build unless user passes -DBUNDLE_SKIP_ATLAS=OFF
+
+# turn off optional OOPS toy models
+  option( ENABLE_LORENZ95_MODEL "Build LORENZ95 toy model" OFF )
+  option( ENABLE_QG_MODEL "Build QG toy model" OFF )
+
+  ecbuild_bundle( PROJECT eckit GIT "https://github.com/ecmwf/eckit.git" TAG 1.16.0 )
+  ecbuild_bundle( PROJECT fckit GIT "https://github.com/ecmwf/fckit.git" TAG 0.9.2 )
+  ecbuild_bundle( PROJECT atlas GIT "https://github.com/ecmwf/atlas.git" TAG 0.35.0 )
+
+# External (required) observation operators
+  # TODO remove the CRTM from here and use it as a library
+  #option("BUNDLE_SKIP_CRTM" "Don't build CRTM" "OFF") # Don't build crtm unless user passes -DBUNDLE_SKIP_CRTM=OFF
+  #ecbuild_bundle( PROJECT crtm GIT "https://github.com/JCSDA/crtm.git"   TAG v2.4.1-jedi.1 )
+  ecbuild_bundle( PROJECT crtm SOURCE "../sorc/crtm" )
+
+# Build GSI-B
+  option(BUILD_GSIBEC "Build GSI-B" OFF)
```

Comment:
Not sure this works for regional apps

Reply:
Yes. seems GSIBEC building failed in my tries.